src/StringWrapper/Iconv.php

```diff
@@ -244,7 +244,12 @@ public function strlen($str)
      */
     public function substr($str, $offset = 0, $length = null)
     {
-        return iconv_substr($str, $offset, $length, $this->getEncoding());
+        return iconv_substr(
+            $str,
+            $offset,
+            $length === null ? iconv_strlen($str, $this->getEncoding()) : $length,
```

Comment:
Btw, could use `$this->strlen($str)` here, to avoid some repetition (and to re-use tested primitives)